app/controllers/membership_applications_controller.rb

```diff
@@ -126,20 +138,32 @@ def authorize_membership_application
   end
 
 
-  def new_upload_file(upload_file_param)
-    if upload_file_param['actual_files']
-      upload_file_param['actual_files'].each do |upload_file|
+  def new_file_uploaded(params)
+
+    successful = true
+
+    if params.fetch('uploaded_file', false) && params['uploaded_file'].fetch('actual_files', false)
+
+      params['uploaded_file']['actual_files'].each do |uploaded_file|
```

Comment:
Yes, you're right about needing a second check on whether the ```actual_files``` sub hash exists before calling ```each``` on it (to be honest, I have not looked at the form to see how it handles that).  To fix that I might do:
```
if (uploaded_files = params['uploaded_file'])

  uploaded_files['actual_files']&.each do |uploaded_file|
```
That is less code and read easier (at least for me).   Using the ```fetch``` tests means that we're confirming the presence of the key, and fetching the value, and (if true) subsequently fetching the value _again_.

However, this is your code and whatever works best for you is what we should go with.